wikidatarefisland/pipes/item_statistical_matcher_pipe.py

```diff
@@ -0,0 +1,93 @@
+from _collections import defaultdict
+
+from wikidatarefisland.pipes import AbstractPipe
+
+
+class ItemStatisticalAnalysisPipe(AbstractPipe):
+    """Pipe to analyze and build a mapping of item qids to values given in the scraped data."""
+
+    def __init__(self, whitelisted_external_identifiers, minimum_repetitions, maximum_noise):
+        """Instantiate the pipe
+
+        :type whitelisted_external_identifiers: list of whitelisted external identifiers
+            like ['P1', 'P3', ...]
+        :type minimum_repetitions: int minimum number of receptions to consider two values the same
+            The higher the number, the higher precision and the lower recall will be
+        """
+        self.whitelisted_external_identifiers = whitelisted_external_identifiers
+        self.minimum_repetitions = minimum_repetitions
+        self.maximum_noise = maximum_noise
+        self.statistics = defaultdict(dict)
+
+    def flow(self, potential_match):
+        """
+        Observe and take notes of items passing by
+
+        This pipe doesn't produce any match on its own
+        but it only builds mapping of most used items and their values in the scraped data.
+        For example if a website uses "(Germany)" as the value of nationality of a person
+        and we have "Q183" as that value and it gets repeated more than N times,
+        we consider these two values to be the same.
+
+        :param potential_match: a line of output of pipe2
+        """
+        if potential_match['statement']['datatype'] != 'wikibase-item':
+            return [potential_match]
+        item_id = potential_match['statement']['value']['numeric-id']
+        ext_id_property = None
+        for pid in potential_match['reference']['referenceMetadata']:
+            if pid not in self.whitelisted_external_identifiers:
+                continue
+            ext_id_property = pid
+            break
+        if not ext_id_property:
+            return [potential_match]
```

Comment:
IMHO it will be worthwhile to change the structure of reference blob to include a pid key (same as it's supposed to include url). This has 2 advantages: 
1) We will not have to make this iteration every time we want to get the pid
2) This class will not have to depend on th whitelisted exernal identifiers anymore, as we already filter these out in Pipe 1.

Reply:
Yes, I would love to have something like that but I would say it's not part of scope of this PR. We should file a bug, track it and pick it up soon

Reply:
Filed: https://phabricator.wikimedia.org/T252718